tests/unit/restapi/v1/test_tags.py

```diff
@@ -0,0 +1,413 @@
+            tag does not match the expected name.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_TAGS_ROUTE}/{tag_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json()["name"] == expected_name
+
+
+def assert_tag_is_not_found(
+    client: FlaskClient,
+    tag_id: int,
+) -> None:
+    """Assert that a tag is not found.
+
+    Args:
+        client: The Flask test client.
+        tag_id: The id of the tag to retrieve.
+
+    Raises:
+        AssertionError: If the response status code is not 404.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_TAGS_ROUTE}/{tag_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 404
+
+
+def assert_cannot_rename_tag_with_existing_name(
+    client: FlaskClient,
+    tag_id: int,
+    existing_name: str,
+) -> None:
+    """Assert that renaming a tag with an existing name fails.
+
+    Args:
+        client: The Flask test client.
+        tag_id: The id of the tag to rename.
+        name: The name of an existing tag.
+
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = rename_tag(
+        client=client,
+        tag_id=tag_id,
+        new_name=existing_name,
+    )
+    assert response.status_code == 400
+
+
+# -- Tests -----------------------------------------------------------------------------
+
+
+@pytest.mark.v1
+def test_create_tag(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+) -> None:
+    """Test that tags can be registered and retrieved using the API.
+
+    This test validates the following sequence of actions:
+
+    - A user registers a tag: {name: "tag"}
+    - The user is able to retrieve information about the tag using the
+      tag id.
+    - The returned information matches the information that was provided
+      during registration.
+    """
+    name = "tag"
+
+    tag_response = actions.register_tag(
+        client,
+        name=name,
+        group_id=auth_account["default_group_id"],
+    )
+    tag_expected = tag_response.get_json()
+
+    assert_tag_response_contents_matches_expectations(
+        response=tag_expected,
+        expected_contents={
+            "name": name,
+            "user_id": auth_account["user_id"],
+            "group_id": auth_account["default_group_id"],
+        },
+    )
+
+    assert_retrieving_tag_by_id_works(
+        client, tag_id=tag_expected["tagId"], expected=tag_expected
+    )
+
+
+@pytest.mark.v1
+def test_tag_search_query(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_tags: dict[str, Any],
+) -> None:
+    """Test that tags can be registered and retrieved using the API with search terms.
+
+    This test validates the following sequence of actions:
+
+    - A user registers three tags: "tag1", "tag2", "tag3".
+    - The user is able to retrieve a list of all registered tags with ...
```

Comment:
Documentation: The line looks unfinished, need to replace `...` with something specific.

Reply:
Fixed in commit: [d6b7e43](https://github.com/usnistgov/dioptra/pull/455/commits/d6b7e4340aac661585c95919bcb0095b00d53854)